src/main/scala/io/github/spark_redshift_community/spark/redshift/RedshiftCatalog.scala

```diff
@@ -0,0 +1,68 @@
+package io.github.spark_redshift_community.spark.redshift;
+
+import java.sql.SQLException
+import java.util
+
+import scala.collection.JavaConverters.mapAsScalaMapConverter
+import scala.collection.JavaConverters.mapAsJavaMap
+
+import io.github.spark_redshift_community.spark.redshift.v2.{RedshiftDataSourceV2}
+import org.apache.spark.sql.connector.catalog.{Identifier, TableChange, Table => SparkTable}
+import org.apache.spark.sql.connector.expressions.Transform
+import org.apache.spark.sql.errors.QueryCompilationErrors
+import org.apache.spark.sql.execution.datasources.jdbc.JDBCOptions
+import org.apache.spark.sql.execution.datasources.v2.jdbc.JDBCTableCatalog
+import org.apache.spark.sql.jdbc.{JdbcDialect, JdbcDialects}
+import org.apache.spark.sql.types.StructType
+import org.apache.spark.sql.util.CaseInsensitiveStringMap
+import io.github.spark_redshift_community.spark.redshift
+
+
+class RedshiftCatalog extends JDBCTableCatalog {
```

Comment:
can we add any test coverage of the catalog capabilities?

Reply:
Yeah catalog is a good way to test the whole thing